src/main/java/org/squiddev/plethora/gameplay/modules/glasses/objects/object2d/Text.java

```diff
@@ -101,8 +147,30 @@ public void draw2D() {
 		GlStateManager.pushMatrix();
 		GlStateManager.translate(position.x, position.y, 0);
 		GlStateManager.scale(size, size, 1);
-		// We use 0xRRGGBBAA, but the font renderer expects 0xAARRGGBB, so we rotate the bits
-		fontrenderer.drawString(text, 0, 0, Integer.rotateRight(colour, 8));
+
+		int x = 0;
+		int y = 0;
+		for (String[] fullLine : lines) {
+			for (String tabSection : fullLine) {
+				// We use 0xRRGGBBAA, but the font renderer expects 0xAARRGGBB, so we rotate the bits
+				if (dropShadow)
+					x = fontrenderer.drawStringWithShadow(tabSection, x, y, Integer.rotateRight(colour, 8));
+				else
+					x = fontrenderer.drawString(tabSection, x, y, Integer.rotateRight(colour, 8));
+				x = (int) Math.floor(x/TAB_WIDTH)*TAB_WIDTH+TAB_WIDTH;
+			}
+			// Carriage return
+			x = 0;
+			// Set x to the next tab location
+			y += this.lineHeight;
+		}
+
 		GlStateManager.popMatrix();
 	}
+
+	private String[][] splitText(String text) {
+		String[] lines = text.split("\n|\r");
```

Comment:
Possibly should be `\r|\n|\r\n`? Though I'm tempted to say just use `\n`. It might be worth pre-constructing the regex as a static and reusing it here.